pandas/tests/series/test_operators.py

```diff
@@ -275,41 +274,24 @@ def test_logical_ops_with_index(self, op):
         assert_series_equal(result, expected)
 
     @pytest.mark.parametrize(
-        "op",
+        "op, index_op",
         [
-            pytest.param(
-                ops.rand_,
-                marks=pytest.mark.xfail(
-                    reason="GH#22092 Index implementation returns Index",
-                    raises=AssertionError,
-                    strict=True,
-                ),
-            ),
-            pytest.param(
-                ops.ror_,
-                marks=pytest.mark.xfail(
-                    reason="Index.get_indexer with non unique index",
-                    raises=InvalidIndexError,
-                    strict=True,
-                ),
-            ),
-            ops.rxor,
```

Comment:
why is this change needed?  I guess the ror_ reason would no longer be accurate, but this doesn't change the fact that `op(index, series)` doesn't behave like `op(series, index)`, which is why we have these xfailed

Reply:
@batterseapower can you respond to this here

Reply:
OK, the intention of the test was not clear. Perhaps it would have been better to have an explicit `isinstance(result, pd.Series)` check in the test rather than having the test explicitly check that `result` was an Index, which made it look like that was the expected behaviour.

Reply:
I've made the tests clearer and restored the xfail mark.